widget/check.go

```diff
@@ -167,8 +176,44 @@ func NewCheck(label string, changed func(bool)) *Check {
 		label,
 		false,
 		changed,
+		false,
 	}
 
 	Renderer(c).Layout(c.MinSize())
 	return c
 }
+
+// FocusGained is called when the Check has been given focus.
+func (c *Check) FocusGained() {
+	if c.disabled {
+		return
+	}
+	c.focused = true
+
+	Refresh(c)
+}
+
+// FocusLost is called when the Check has had focus removed.
+func (c *Check) FocusLost() {
+	c.focused = false
+
+	Refresh(c)
+}
+
+// Focused returns whether or not this Check has focus.
+func (c *Check) Focused() bool {
+	if c.disabled {
+		return false
+	}
+	return c.focused
+}
+
+// TypedRune receives text input events when the Check is focused.
+func (c *Check) TypedRune(r rune) {
+	if r == 32 { //space
```

Comment:
Please use ' ' instead of 32 as it does not need the comment...

Reply:
There is currently an issue whereby this does not check if disabled.
Whilst that should never trigger I suspect that we should guard it here too.